src/ILVerify/src/TypeSystemHelpers.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem;
+
+namespace Internal.IL
+{
+    public static class TypeSystemHelpers
```

Comment:
We have this in `src\Common\src\TypeSystem\Common\TypeSystemConstaintsHelpers.cs` already. ILVerify should use the existing impl.

Reply:
Well, that typo in the filename cost me some time, haha. Might be worth fixing in another PR.

The mentioned helper class does not cover the case where a special constrained is set and the instantiation is a generic parameter itself (with for instance the same special constraint).
E.g. this valid C# code will report a verification error:

```c-sharp
public class Constrained<T> where T : class { }
...

public void InstantiateGenerc<T>() where T : class
{
    Constrained<T> c = new Constrained<T>();
}
```
The test case "Casting.AssignReturnValueFromConstrainedProperty" will also fail because of this.
I will port the relevant part from my implementation and add another test case for this scenario.

Reply:
>  Might be worth fixing in another PR.

Feel free to include the fix it in this PR too if you would like.

Reply:
Ok, thanks. I will do so.